util/binfmt_misc/detect.go

```diff
@@ -14,19 +15,53 @@ func SupportedPlatforms() []string {
 	once.Do(func() {
 		def := platforms.DefaultString()
 		arr = append(arr, def)
-
-		if p := "linux/amd64"; def != p && amd64Supported() {
+		if p := "linux/amd64"; def != p && amd64Supported() == nil {
 			arr = append(arr, p)
 		}
-		if p := "linux/arm64"; def != p && arm64Supported() {
+		if p := "linux/arm64"; def != p && arm64Supported() == nil {
 			arr = append(arr, p)
 		}
-		if !strings.HasPrefix(def, "linux/arm/") && armSupported() {
+		if !strings.HasPrefix(def, "linux/arm/") && armSupported() == nil {
 			arr = append(arr, "linux/arm/v7", "linux/arm/v6")
 		} else if def == "linux/arm/v7" {
 			arr = append(arr, "linux/arm/v6")
 		}
 	})
-
 	return arr
 }
+
+//Here, the func just validate the platforms and show warning message if there is,
+//the end user could fix the issue based on those warning, and thus no need to drop
+//the platfrom from the candidates.
+func WarnIfUnsupported(pfs []string) {
```

Comment:
(Sorry, didn't notice this before.) The godoc convention is that a function comment needs to start with the function name. Eg. `// WarnIfUnsupported validates the platform and shows warning ....`

Reply:
np, you left many good comments! really learnt a lot from those comments, thanks!